client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -1,58 +1,68 @@
-	i18n = require( 'i18n-calypso' ),
-	Shortcode = require( 'lib/shortcode' ),
-	closest = require( 'component-closest' );
+import ReactDom from 'react-dom';
+import ReactDomServer from 'react-dom/server';
+import { Provider as ReduxProvider } from 'react-redux';
+import React from 'react';
+import tinymce from 'tinymce/tinymce';
+import { assign, debounce, find, findLast, pick, values } from 'lodash';
+import i18n from 'i18n-calypso';
+import Shortcode from 'lib/shortcode';
+import closest from 'component-closest';
 
 /**
  * Internal dependencies
  */
-var sites = require( 'lib/sites-list' )(),
-	PostActions = require( 'lib/posts/actions' ),
-	PostEditStore = require( 'lib/posts/post-edit-store' ),
-	MediaConstants = require( 'lib/media/constants' ),
-	MediaActions = require( 'lib/media/actions' ),
-	MediaUtils = require( 'lib/media/utils' ),
-	MediaSerialization = require( 'lib/media-serialization' ),
-	MediaMarkup = require( 'post-editor/media-modal/markup' ),
-	MediaModalViews = require( 'post-editor/media-modal/constants' ).Views,
-	MediaStore = require( 'lib/media/store' ),
-	MediaLibrarySelectedData = require( 'components/data/media-library-selected-data' ),
-	EditorMediaModal = require( 'post-editor/media-modal' ),
-	notices = require( 'notices' ),
-	TinyMCEDropZone = require( './drop-zone' ),
-	restrictSize = require( './restrict-size' ).default,
-	advanced = require( './advanced' ),
-	Gridicon = require( 'components/gridicon' ),
-	config = require( 'config' );
+import SiteListFactory from 'lib/sites-list';
+const sites = SiteListFactory();
+import PostActions from 'lib/posts/actions';
+import PostEditStore from 'lib/posts/post-edit-store';
+import MediaConstants from 'lib/media/constants';
+import MediaActions from 'lib/media/actions';
+import MediaUtils from 'lib/media/utils';
+import { deserialize } from 'lib/media-serialization';
+import MediaMarkup from 'post-editor/media-modal/markup';
+import { Views as MediaModalViews } from 'post-editor/media-modal/constants';
+import MediaStore from 'lib/media/store';
+import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
+import EditorMediaModal from 'post-editor/media-modal';
+import notices from 'notices';
+import TinyMCEDropZone from './drop-zone';
+import restrictSize from './restrict-size';
+import advanced from './advanced';
+import Gridicon from 'components/gridicon';
+import config from 'config';
+import { getSelectedSite } from 'state/ui/selectors';
 
 /**
  * Module variables
  */
-var REGEXP_IMG = /<img\s[^>]*\/?>/ig,
+const REGEXP_IMG = /<img\s[^>]*\/?>/ig,
 	SIZE_ORDER = [ 'thumbnail', 'medium', 'large', 'full' ];
 
 function mediaButton( editor ) {
-	var nodes = {},
-		updateMedia, resizeEditor;
+	const store = editor.getParam( 'redux_store' );
+	if ( ! store ) {
+		return;
+	}
+
+	const { getState, subscribe } = store;
+
+	let nodes = {},
+		resizeEditor,
+		selectedSite = getSelectedSite( getState() ),
+		updateMedia;
```

Comment:
It's not possible in this case because of the order of usage, loadedImages uses updateMedia which uses loadedImages and since const can not be declared before assigned, I kept it this way. A more profound change to these functions can probably fix this, but I think this PR is already big enough.